library/common-library/templates/_images.tpl

```diff
@@ -0,0 +1,53 @@
+{{/*
+Return the proper image name
+{{ include "common.images.image" ( dict "imageRoot" .Values.path.to.the.image "context" .) }}
+*/}}
+{{- define "common.images.image" -}}
+{{- $registryName := "" -}}
+{{- $repositoryName := .imageRoot.repository -}}
+{{- $tag := .imageRoot.tag | default .context.Chart.AppVersion | toString -}}
+
+{{- if .context.Values.global }}
+    {{- if .context.Values.global.image.registry }}
+     {{- $registryName = .context.Values.global.image.registry  -}}
+    {{- end -}}
+{{- end -}}
+
+{{- with .imageRoot.registry }}
+    {{- $registryName = . -}}
+{{ end }}
```

Comment:
Create more than one helper. For registry, image and tag:
 - The registry needs a global but image and tag don't.
 - Assume that the image is going to be in `.Values.image.repository`. We know that we are going to have sidecars but in that case we can override the helper that returns the image and add other helper for the sidecar.
 - Do not use the same helper for all the `registry/image:tag` so it can over used/overriden in parts.
 - You can then have other helper that use the three written above to create the full image path so it is readable in the deployment.yaml

Reply:
I don't think we can assume that the image is going to be in `.Values.image.repository`. Most of charts uses more than one image, either for the integration, the webhook, the agent.

Reply:
Then we should do common helpers for what is common: registry. The rest should be done in each chart.

Reply:
We discussed about this PR offline, we agreed that:
 - yes  image and tag do not need global values
 - we cannot assume image is in the same root in different repos -> most of charts uses different images
 - we need to split the helpers
 - sadly the complex function signature is needed. If too complex we can decide to drop that "global"